### PR TITLE
BC-4422 - Whitelist frame ancestors

### PR DESCRIPTION
**Pull request by uidp**
# Description
Enhance CSP-configuration to prevent ClickJacking attacks.

## Links to Tickets or other pull requests
Ticket [BC-4422](https://ticketsystem.dbildungscloud.de/browse/BC-4422)
https://github.com/hpi-schul-cloud/dof_app_deploy/pull/576

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by Loki-Afro — APPROVED** [review]

**Review by virgilchiriac — APPROVED** [review]

File: ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2 [hunk 1]
```diff
@@ -26,7 +26,7 @@ data:
       Use:
         defaultSrc, fontSrc, styleSrc, scriptSrc, imageSrc,
         connectSrc, mediaSrc, objectSrc, prefetchSrc, childSrc,
-        frameSrc, workerSrc, frameancestorsSrc, formactionSrc, baseuriSrc
+        frameSrc, workerSrc, frameAncestors, formactionSrc, baseuriSrc
```
Comment (virgilchiriac) [inline ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2:29]: do we need to update also the reference to 
https://acc.bettermarks.com in line 65 below?

File: ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2 [hunk 2]
```diff
@@ -50,7 +51,7 @@ data:
             use * as value for
             defaultSrc, fontSrc, styleSrc, scriptSrc, imageSrc,
             connectSrc, mediaSrc, objectSrc, prefetchSrc, childSrc,
-            frameSrc, workerSrc, frameancestorsSrc, formactionSrc, baseuriSrc
+            frameSrc, workerSrc, frameAncestors, formactionSrc, baseuriSrc
```
Comment (virgilchiriac) [inline ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2:54]: same question here, if we need to update https://acc.bettermarks.com reference below

Comment (Loki-Afro) [reply ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2:54]: no

File: ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2 (hunk 1 above)
Comment (Loki-Afro) [reply ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2:29]: no

**Review by virgilchiriac — APPROVED** [review]

**Review by Loki-Afro — APPROVED** [review]

**Pull request merged**